packages/connect/src/auth.ts

```diff
@@ -33,6 +33,26 @@ export interface AuthOptions {
   };
 }
 
+/**
+ * Firefox and mobile should not use a 'popup' type of window.
+ */
+export const shouldUsePopup = () => {
+  const isMobile = !!('onTouchStart' in document.documentElement);
```

Comment:
do we need a `typeof document !== 'undefined'` here, for ssr apps that include this?

Reply:
I don't think so, because `authenticate` already relies on a client-side environment (i.e. `window.open`).